contribute/pull-request.md

```diff
@@ -0,0 +1,94 @@
+- To avoid regressions, the pull request should include tests that replicate the fixed bug.
+
+### Frontend-specific guidelines
+
+Pull requests for frontend contributions must:
+
+- Use [Emotion](https://github.com/grafana/grafana/blob/master/style_guides/styling.md) for styling.
+- Not increase the Angular code base.
+- Not use `any` or `{}` without comments describing why.
+- Not contain large React components that could easily be split into several smaller components.
+- Not contain backend calls directly from components—use actions and Redux instead.
+
+Pull requests for Redux contributions must:
+
+- Use the `actionCreatorFactory` and `reducerFactory` helpers instead of traditional switch statement reducers in Redux. Refer to [Redux framework](https://github.com/grafana/grafana/tree/master/style_guides/redux.md) for more details.
+- Use `reducerTester` to test reducers. Refer to [Redux framework](https://github.com/grafana/grafana/tree/master/style_guides/redux.md) for more details.
+- Not contain code that mutates state in reducers or thunks.
+- Not contain code that accesses the reducers state slice directly. Instead, the code should use state selectors to access state.
+
+## Code review
+
+Once you've created a pull request, the next step is to have someone review your change. A review is a learning opportunity for both the reviewer and the author of the pull request.
+
+If you think a specific person needs to review your pull request, then you can tag them in the description or in a comment. Tag a user by typing the `@` symbol followed by their GitHub username.
+
+We recommend that you read [How to do a code review](https://google.github.io/eng-practices/review/reviewer/) to learn more about some best practices for code reviews.
+
+## Formatting guidelines
+
+A well-written pull request minimizes the time to get your change accepted. These guidelines help you write good commit messages and descriptions for your pull requests.
+
+### Commit message format
+
+Grafana uses the guidelines for commit messages outlined in [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/), with the following additions:
+
+- Subject line must begin with the _area_ of the commit.
+- A footer in the form of an optional [keyword and issue reference](https://help.github.com/en/articles/closing-issues-using-keywords).
+
+#### Area
+
+The area should use upper camel case, e.g. UpperCamelCase.
+
+Prefer using one of the following areas:
+
+- **Build:** Changes to the build system, or external dependencies.
+- **Chore:** Changes that don't affect functionality.
+- **Dashboard:** Changes to the Dashboard feature.
+- **Docs:** Changes to documentation.
+- **Explore:** Changes to the Explore feature.
+- **Plugins:** Changes to any of the plugins.
+
+For changes to data sources, the area should be the name of the data source, e.g., AzureMonitor, Graphite, and Prometheus.
```

Comment:
```suggestion
For changes to data sources, the area should be the name of the data source, e.g. AzureMonitor, Graphite, and Prometheus.
```

Reply:
The Chicago Manual of Style appears to say otherwise :) I think without is preferred in British writing?

Reply:
I believe that rule is only for citations and not examples. Could you point me in the direction to find it?

The only example I could find where my assumption holds is [this one](https://www.chicagomanualofstyle.org/qanda/data/faq/topics/Abbreviations/faq0057.html). This is mostly for my understanding as I'm not really fuzzed if you would like to leave my change out and by no means would like such a small thing to block this PR.

Reply:
Actually, looking at the Microsoft Style Guide, they advise not using e.g. at all:

> Don't use. Use for example, such as, or like, as appropriate.

Based on that, I'd prefer to this rewrite to:

```
For changes to data sources, the area should be the name of the data source, such as AzureMonitor, Graphite, and Prometheus.
```

What do you think?

Reply:
I can agree that this _particular_ piece reads well, but that's one that I'm willing to fight for. 

I like the use of `e.g.` too much as it seems incredibly simple and concise from my POV.

Reply:
and by this piece I meant the text, you suggested.

Reply:
I'm a fan too, so let's leave it as for now. Though we should consider adding any deviation from the Microsoft Style Guide to our own docs style guide. I'll keep this in mind!